src/frontend/js/common/searchFields/index.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { AutosuggestProps } from 'react-autosuggest';
+
+import { APICourseSearchResponse } from '../../types/api';
+import { Suggestion, SuggestionSection } from '../../types/Suggestion';
+import { handle } from '../../utils/errors/handle';
+import { getSuggestionsSection } from './getSuggestionsSection';
+
+/**
+ * Define the kind of suggestions our `<SearchSuggestField />` component supports.
+ */
+type SearchSuggestion = Suggestion<'categories' | 'organizations' | 'persons'>;
+
+/**
+ * Derive from `SearchSuggestion` the kind of suggestion sections our `<SearchSuggestField />` component supports.
+ */
+export type SearchSuggestionSection = SuggestionSection<SearchSuggestion>;
+
+/**
+ * Helper to typecheck `react-autosuggest` expected props. Defines what is acceptable as a suggestion
+ * throughout our code related to this instance of `<Autosuggest />`.
+ */
+export type SearchAutosuggestProps = AutosuggestProps<SearchSuggestion>;
+
+/**
+ * `react-autosuggest` callback to get a human string value from a Suggestion object.
+ * @param suggestion The relevant suggestion object.
+ */
+export const getSuggestionValue: SearchAutosuggestProps['getSuggestionValue'] = suggestion =>
+  suggestion.title;
+
+/**
+ * `react-autosuggest` callback to render one suggestion.
+ * @param suggestion Either a resource suggestion with a model name & a machine name, or the default
+ * suggestion with some text to render.
+ */
+export const renderSuggestion: SearchAutosuggestProps['renderSuggestion'] = suggestion => (
```

Comment:
Do you consider this function as a react component ? If not, I think you don't need to import React in this module and as ther is no jsx you can rename the file in `.ts`

Reply:
This function uses JSX and therefore needs `.tsx` and `import React`. Are you suggesting changes?

Reply:
No forget my comment 🙂